cmake_modules/commonSetup.cmake

```diff
@@ -378,6 +384,20 @@ IF ("${COMMONSETUP_DONE}" STREQUAL "")
            add_definitions(/J)
         endif ()
     endif ()
+
+
+    #===  WARNING ======
+    # Temporary disable warnings. Reenable them one by one to reexamine and fix them.
+    IF (NOT ALL_WARNINGS_ON)
+      set (WARNINGS_IGNORE "/wd4267 /wd4244 /wd6340 /wd6297 /wd4018 /wd4302 /wd4311 /wd4320 /wd4800") # data conversion warnings
+      set (WARNINGS_IGNORE "${WARNINGS_IGNORE} /wd4251 /wd4275") # dll-interface for used by clients
+      set (WARNINGS_IGNORE "${WARNINGS_IGNORE} /wd6246")   # local variable hidden by outter scope
+      set (WARNINGS_IGNORE "${WARNINGS_IGNORE} /wd6031")   # Return value ignored
+      set (WARNINGS_IGNORE "${WARNINGS_IGNORE} /wd4005")   # MACRO redef: same value
+      set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${WARNINGS_IGNORE}")
```

Comment:
What about release builds?

Some of these warnings seem useful - what is the plan for re-enabling them?

Reply:
Good point. Using CMAKE_CXX_FLAGS instead will work for all build types.
We can open one ticket per warning to re-enable them.